src/frontend/mame/luaengine.cpp

```diff
@@ -1350,6 +1351,7 @@ void lua_engine::initialize()
 						table[rom.name()] = rom;
 				return table;
 			});
+	device_type["cassette"] = sol::property([] (device_t &dev) { return dynamic_cast<cassette_image_device *>(&dev); });
```

Comment:
I’d prefer if this was done by exposing another type of device enumerator, like the `screen_enumerator` for finding/enumerating screen devices.